tests/e2e/idps_test.go

```diff
@@ -0,0 +1,77 @@
+package e2e
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	conn "github.com/terraform-redhat/terraform-provider-rhcs/tests/ci"
+	"github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/cms"
+	con "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/constants"
+	exe "github.com/terraform-redhat/terraform-provider-rhcs/tests/utils/exec"
+)
+
+var _ = Describe("TF Test", func() {
+	Describe("Identity Providers test cases", func() {
+		var clusterID string
+		var idpServices map[string]*exe.IDPService
+		var (
+			htpasswdUsername = "jacko"
+			htpasswdPassword = "1q2wFe4rpoe2318"
+		)
+		htpasswdMap := []interface{}{map[string]string{"username": htpasswdUsername, "password": htpasswdPassword}}
+
+		BeforeEach(func() {
+			clusterID = os.Getenv("CLUSTER_ID")
+			// for future use
+			idpServices = map[string]*exe.IDPService{
```

Comment:
nit, Why the use of a map? I would suggest creating a **struct** to hold these services. This way, I won't have to address a service with a string that might "change". Instead of 
`idpSevices["service-x"]` it would simply be `idpServices.ServiceX`.

It would be easier to maintain. Plus, we won't add to the map entries during runtime. I think...

Reply:
tnx! fixed :D